### PR TITLE
Implement context manager partial template test case

### PR DESCRIPTION
**Pull request by asottile**

### CODE REVIEW

**Comment by struys** [comment]
lg2m

File: Cheetah/testing/partial_template_test_case.py [hunk 1]
```diff
@@ -5,9 +5,20 @@
 
 import pyquery
 
+from Cheetah.DummyTransaction import DummyTransaction
 from Cheetah.Template import Template
 
 
+_KNOWN_BASE_TESTS = set()
```
Comment (bukzor) [inline Cheetah/testing/partial_template_test_case.py]: module-scope mutables are madness.

File: Cheetah/testing/all_partials_tested.py [hunk 2]
```diff
@@ -84,7 +85,7 @@ def get_partial_methods(template_packages):
 def is_partial_test_cls(cls):
     return (
         issubclass(cls, PartialTemplateTestCase) and
-        cls is not PartialTemplateTestCase and
+        cls not in _KNOWN_BASE_TESTS and
         cls.partial is not None and
         cls.method is not None
     )
```
Comment (bukzor) [inline Cheetah/testing/all_partials_tested.py]: Figure out some duck-typing way to do this that you're happy with.

How about `and cls.__dict__.get('__test__', True)`?

File: Cheetah/testing/partial_template_test_case.py (hunk 1 above)
Comment (asottile) [inline Cheetah/testing/partial_template_test_case.py]: this is only for tests

File: Cheetah/testing/all_partials_tested.py (hunk 2 above)
Comment (bukzor) [inline Cheetah/testing/all_partials_tested.py]: ^

**Pull request merged**